src/query-builder/function-module.ts

```diff
@@ -142,14 +145,19 @@ export class FunctionModule<DB, TB extends keyof DB> {
    * ```
    */
   max<
-    O extends number | string | bigint,
-    C extends SimpleReferenceExpression<DB, TB> = DynamicReferenceBuilder<any>
+    O extends number | string | bigint | null = number | string | bigint,
+    C extends SimpleReferenceExpression<DB, TB> = SimpleReferenceExpression<
+      DB,
+      TB
+    >
   >(
     column: C
   ): AggregateFunctionBuilder<
     DB,
     TB,
-    ExtractTypeFromReferenceExpression<DB, TB, C, O>
+    ExtractTypeFromReferenceExpression<DB, TB, C, O> | null extends O
+      ? null
+      : never
```

Comment:
Current types allow consumers to state `O` only when column is a `DynamicReference`. Let's open this up, so the consumer can tell us whether `O` is nullable, even when column is a `StringReference`, but let's still restrict return type to the type provided for that column in `Database`.